### DISCUSSION TITLE
Extending the library to the other versions of ICD 10

### DISCUSSION THREAD

**Discussion by peymvn** [StefanoTrv/simple_icd_10_CM#7] [Ideas]
Hi Stefano,
Thanks for your wonderful package. It is one of the most extensive and well-designed health projects I've ever seen.
I was wondering how to extend this library to other versions of ICD 10 (in my case Canadian version). I have a list of diseases like the .txt file you have, but unfortunately, there is nothing like the .xml file for the hierarchical relations between diseases. Do you have any recommendations for integrating ICD-10 CA into this library easily?

**Comment by StefanoTrv** [comment]
Hello, thank you for your compliments.
Sadly, I don't think there's an easy way to do it.
In an old version of my other library [simple_icd_10](https://github.com/StefanoTrv/simple_icd_10), I had two lists that contained all the codes and their descriptions and then I managed manually all the unusual blocks ([here the old implementation](https://github.com/StefanoTrv/simple_icd_10/blob/e63d19a1bc570f3ef8df0f6bbcf27d652c33c808/simple_icd_10.py)). There probably was a better way to deal with them, but in the end I created an XML file using the library ([with this code](https://github.com/StefanoTrv/simple_icd_10/blob/master/other/XML%20file%20generator.py)) and now simple_icd_10, like this library, works using a XML file (simpler than the one in this library).
However, I don't know how much this old code can be of help, since it only deals with the standard four-character long ICD-10 codes.
I read [here](http://mchp-appserv.cpe.umanitoba.ca/viewDefinition.php?printer=Y&definitionID=102937) that the classifications are identical up the fourth character; assuming this is true I would:

1. use the old code for simple_icd_10 to create a new XML file that contains the same old codes with the new descriptions. Basically you load into the two lists the rows of your .txt file, except for the ones that contain the longer codes.
2. starting from the new XML file, programmatically modify the XML data by adding the remaining codes.
3. take the code of one of my libraries and modify it to make it work for your needs. Just remember that the two libraries use slightly different XML formats, so you'll have to slightly modify "XML file generator.py" if you want to use simple_icd_10_CM as your base code.

I hope this helps!

**Reply by peymvn** [reply]
Thanks a lot!
